api/src/test/java/org/openmrs/CohortTest.java

```diff
@@ -165,4 +168,127 @@ public void setMemberIds_shouldSupportLargeCohorts() {
         double secondsToSet = (endTime - startTime)/1000;
         Assert.assertTrue("Setting cohort of size " + cohortSize + " took " + secondsToSet + " seconds", secondsToSet < 5);
     }
+    
+    @Test
+	public void hasActiveMembership_shouldFindAnActiveMemberCorrectly() throws Exception{
+	
+	    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+	    Date endDateEarlier = dateFormat.parse("2007-02-01 00:00:00");
+	
+	    Calendar calendar = Calendar.getInstance();
+	    calendar.add(Calendar.DAY_OF_YEAR, 1);
```

Comment:
If the earlier date is 2007, can you explain what value the above line adds?

Reply:
What happens if you do not add 1 to today?

Reply:
The test will fail. I can explain it to you. The endDate will be set to today including the current miliseconds. Then at the point hasActiveMembership() is called it will execute  the function isActive() of the class CohortMembership. That method looks like the following:

`Date date = asOfDate == null ? new Date() : asOfDate;
		return !this.getVoided() && OpenmrsUtil.compare(startDate, date) <= 0
				&& OpenmrsUtil.compareWithNullAsLatest(date, endDate) <= 0;`

That method set the reference day to the current date (by using new Date()) which would be a later moment compared to endDateLater and return false. Then the follwoing test case would fail:

	    assertTrue(cohort.hasActiveMembership(7));


As a result it would not be tested if hasActiveMembership works in a positive case where the endDate is set to the future.

Reply:
And lastly, what would happen if one runs these tests with the system clock set to the year 2005?

Reply:
@dkayiwa I updated all in a way that pastDate will be yesteday and future date will be tomorrow. That should be a lot more stable an independent from the clock time of the system.